src/System.Private.CoreLib/src/System/Number.CoreCLR.cs

```diff
@@ -2,19 +2,733 @@
+    // semicolons. The results of having one, two, or three sections in the format
+    // string are described in the table below.
+    //
+    // Sections:
+    //
+    // One - The format string applies to all values.
+    //
+    // Two - The first section applies to positive values
+    // and zeros, and the second section applies to negative values. If the number
+    // to be formatted is negative, but becomes zero after rounding according to
+    // the format in the second section, then the resulting zero is formatted
+    // according to the first section.
+    //
+    // Three - The first section applies to positive
+    // values, the second section applies to negative values, and the third section
+    // applies to zeros. The second section may be left empty (by having no
+    // characters between the semicolons), in which case the first section applies
+    // to all non-zero values. If the number to be formatted is non-zero, but
+    // becomes zero after rounding according to the format in the first or second
+    // section, then the resulting zero is formatted according to the third
+    // section.
+    //
+    // For both standard and user-defined formatting operations on values of type
+    // float and double, if the value being formatted is a NaN (Not
+    // a Number) or a positive or negative infinity, then regardless of the format
+    // string, the resulting string is given by the NaNSymbol,
+    // PositiveInfinitySymbol, or NegativeInfinitySymbol property of
+    // the NumberFormatInfo used to format the number.
+    //
+    // Parsing
+    //
+    // The Parse methods provided by the numeric classes are all of the form
+    //
+    //  public static XXX Parse(String s);
+    //  public static XXX Parse(String s, int style);
+    //  public static XXX Parse(String s, int style, NumberFormatInfo info);
+    //
+    // where XXX is the name of the particular numeric class. The methods convert a
+    // string to a numeric value. The optional style parameter specifies the
+    // permitted style of the numeric string. It must be a combination of bit flags
+    // from the NumberStyles enumeration. The optional info parameter
+    // specifies the NumberFormatInfo instance to use when parsing the
+    // string. If the info parameter is null or omitted, the numeric
+    // formatting information is obtained from the current culture.
+    //
+    // Numeric strings produced by the Format methods using the Currency,
+    // Decimal, Engineering, Fixed point, General, or Number standard formats
+    // (the C, D, E, F, G, and N format specifiers) are guaranteed to be parseable
+    // by the Parse methods if the NumberStyles.Any style is
+    // specified. Note, however, that the Parse methods do not accept
+    // NaNs or Infinities.
+    //
+    // This class contains only static members and does not need to be serializable 
+
+
     internal static partial class Number
```

Comment:
There are a lot of unrelated changes for `Double` formatting/parsing - that should be separated from `Decimal` changes if possible.
And code flow direction should be from src/X.cs or X.CoreCLR.cs to shared/X.cs, not from CoreRT to unshared CoreCLR code? The goal is not to copy CoreRT code to CoreCLR but to share it and leave the hopefully minor differences in X.CoreCLR.cs/X.CoreRT.cs files.

Reply:
I agree, initially I just wanted to make sure it compiles and works for CoreCLR in general (because as I said we are going to use it in mono), I'll move it to the shared folder.